client/lib/trials/get-migration-trial-checkout-url.ts

```diff
@@ -0,0 +1,16 @@
+import { addQueryArgs } from 'calypso/lib/url';
+
+interface MigrationTrialCheckoutUrlArguments {
+	productSlug: string;
+	siteSlug: string;
+}
+
+export function getMigrationTrialCheckoutUrl( {
+	productSlug,
+	siteSlug,
+}: MigrationTrialCheckoutUrlArguments ): string {
+	return addQueryArgs(
+		{ redirect_to: `/plans/my-plan/trial-upgraded/${ siteSlug }` },
+		`/checkout/${ siteSlug }/${ productSlug }`
+	);
+}
```

Comment:
Since this function is nearly identical to the `get-ecommerce-trial-checkout-url.ts`, should we make this a more general trial function for trial upgrades? Maybe something like `getTrialCheckoutUrl`? Or do you see this will be different compare to the eCommerceTrial one in the near future?